apps/src/music/player/MusicPlayer.ts

```diff
@@ -432,6 +482,56 @@ export default class MusicPlayer {
         results.push(resultEvent);
       }
 
+      return results;
+    } else if (event.type === 'chord') {
+      const chordEvent = event as ChordEvent;
+      const {instrument, notes, playStyle} = chordEvent.value;
+      const results: SampleEvent[] = [];
+
+      const folder: SoundFolder | null =
+        this.library.groups[0].folders.find(
+          folder => folder.path === instrument
+        ) || null;
+
+      if (folder === null) {
+        console.warn(`No instrument ${instrument}`);
+        return [];
+      }
+
+      if (playStyle === 'arpeggio-up') {
+        notes.sort();
+      } else if (playStyle === 'arpeggio-down') {
+        notes.sort().reverse();
+      } else if (playStyle === 'arpeggio-random') {
+        // Randomize using Fisher-Yates Algorithm
+        for (let i = notes.length - 1; i > 0; i--) {
+          const j = Math.floor(Math.random() * (i + 1));
+          const temp = notes[i];
+          notes[i] = notes[j];
+          notes[j] = temp;
+        }
+      }
+
+      for (let i = 0; i < (playStyle === 'together' ? notes.length : 16); i++) {
+        const note = notes[i % notes.length];
+        const sound = folder.sounds.find(sound => sound.note === note) || null;
+        if (sound === null) {
+          console.warn(
+            `No sound for note value ${note} on instrument ${instrument}`
```

Comment:
If I haven't yet selected a sound, I see a lot of this in the developer console; 16 times per block, in fact, which blows out if the block is in a loop.  Not a huge issue, but possibly we could emit this once per block, instead of once per note?

Reply:
Oh good call, I'll add a check for this.